### ISSUE TITLE
Permanent wrong placement of footnotes (left over notes)

### ISSUE THREAD

**Issue by sieversMartin** [maieul/ledmac#598]
DISCLAIMER: I have to admit, that I have not managed to set-up a MWE yet. However, I think the issue is worth asking, what might be the next step towards a solution.

I have a document using the most recent version of reledmac for six series of familiar footnotes. No matter how often I compile this document I regularly get messages like this:

```
reledmac reminder: 
 The number of the footnotes in this section has changed since the last run.
 You will need to run LaTeX two more times before the footnote placement
 and line numbering in this section are correct.
Section 16 (10-mappe-12.16

Package reledmac Warning: Line-list file 10-mappe-12.16 was obsolete. 
We have not read it. Please run LaTeX again. on input line 1.
```

As I said, trying to recompile does not solve the problem. Footnotes are placed far away from their origin page.

**Comment by maieul** [comment]
Disclaimer: without MWE, I can't work.

**Comment by sieversMartin** [comment]
Okay, I will try to produce one. If you have any idea what to do/avoid regarding left over notes please let me know.

**Comment by maieul** [comment]
no, honestly, I have no idea. It could depend of many factor. Are you in parallel typesetting ?

**Comment by maieul** [comment]
I use reledmac every day for my thesis, and I have not this problem. It can happen with specific case, but as I have no idea of all you work, I can't tell…

**Comment by sieversMartin** [comment]
My apologies again for not being able to set-up a MWE. As you may imagine, the project's document class is rather complex and a lot of things play a role when it comes to footnote placement.

However, I managed to hunt down the problem within my concrete example file. It is caused by expansion switches of `xstring`, e.g. `\normalexpandarg` etc. I did not pay enough attention to grouping, which caused following string operations done by `reledmac` to fail or at least work not as expected. The same issue by the way caused an infinite loop some days ago when migrating from `eledmac` to `reledmac` for that specific project.

Of course this is not a bug in `reledmac`. However, maybe you could add a short note in the documentation or even use explicit `xstring` switches yourself in order to avoid side effects.

Thank you for your great work and keep TeXing.

**Comment by maieul** [comment]
If I understand right, you have changed the expansion mechanism of `xstring`, which have perturbated reledmac?

In this case, I think the better should be that I ensure to have the correct expansion setting in any case. However, as that is a bug which happens very rarely, and as the correct could have boarder effect,  and as my time is very limited, I will wait before any investigation in this side.

**Comment by sieversMartin** [comment]
> If I understand right, you have changed the expansion mechanism of `xstring`, which have perturbated reledmac?

Yes, I have to use `normalexpandarg` at some macros and did not group them. Thus, this mode for a lot of `xstring` operations stayed active and was accidentally used by `reledmac` as well causing severe problems.

The default mode is `fullexpandarg`. There is a "save and restore" mechanism as well.

**Comment by maieul** [comment]
Ok. I think I could use the save and restore mechanisme inside reledmac, it would be with the less side board effect.
